### PR TITLE
DM-39804: Update to rubin-env 7.0.0.

### PR DESCRIPTION
**Pull request by ktlim**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by ktlim** [comment]
@conda-forge-admin, please rerender

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -32,18 +32,20 @@ gsl:
   - '2.7'
 libaprutil:
   - '1'
+libcxx:
+  - '16'
```
Comment (erykoff) [inline recipe/conda_build_config.yaml:36]: Wait, I think this should just be osx.

Comment (ktlim) [reply recipe/conda_build_config.yaml:36]: It's `- libcxx {{ libcxx }}.*  # [not aarch64]` over in `meta.yaml`

Comment (ktlim) [reply recipe/conda_build_config.yaml:36]: Is it OK, wasteful, or harmful as is?  I've been building this way on Linux for a while.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -170,6 +167,8 @@ outputs:
         - pytorch
         - pyyaml
         - requests
+        - rucio-clients <=1.29.12
+        - ruff ==0.0.275
```
Comment (timj) [inline recipe/meta.yaml:171]: We've been using 0.0.278 in pre-commit but I see 0.0.276 is the only one on conda-forge at the moment. Don't suppose it matters much.

**Review by timj — APPROVED** [review]
This looks like we agreed.

File: recipe/meta.yaml (hunk 2 above)
Comment (ktlim) [reply recipe/meta.yaml]: Changed to 0.0.278.

**Pull request merged**